File: Problema1/Personajes.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Arrays;
public class Personajes{
	HashMap<String,ArrayList<String>> personajes;
	int ganador;
	final String[][] fisicas={
							{"Azul","Negro","Amarillo","Rojo","Cafe","Violeta","Gris","Verde","Calvo","Rosa"}, //Color de pelo
							{"Delgado","Flaco","Gordo","Morbido","Atletico","Robusto","Muy Atletico","Esqueletico","Normal","Amorfo"}, //Complexión
							{"Alto","Bajo","Enano","Gigante","Godzilla","Liliputiense","Microscopico","Planeta","Subatomico","Enorme"},//Estatura
							{"Blanco","Cafe","Negro","Rojo","Azul","Verde","Morado","Rosa","Amarillo","Carton"},//Color de piel
							{"Humano","Monstruo","Saiyajin","Depredador","Robot","Reptiliano","Atlante","Hombre pez","Alien","Marciano"}//Raza
							};
	final String[][] sobresalientes={
							{"Arquitectura","Cine","Pintura","Escultura","Poesia","Musica","Danza"}, //Artes
							{"Football","Baseball","Basquetball","Soccer","Esports","Rugby","Archery"}, //Deportes
							{"Química","Fisica","Biologia","Computacion","Fisica Cuantica","Epidemiologia","Farmacologia"},//Ciencia
							{"Licenciado","Ingeniero","Maestro","Doctor","Postdoctor","Bachiller","Lider Supremo"},//Grado
							{"Puzzles","Acuarios","Terrarios","Legos","Caminata","Maquetas","Dormir"}//Hobby
							};		
	public Personajes(){
		this.personajes=new HashMap<String,ArrayList<String>>(10);
		generaCaracteristicas();
		ganador=(int)Math.floor(Math.random()*11);
	}

	public void generaCaracteristicas(){
		ArrayList<String> caracteristicas;
		double numero,numero2;
		//Genera un numero random para escoger las caracteristicas
		numero=Math.random();
		for(int i=0;i<10;i++){
			caracteristicas=new ArrayList<String>(5);
			for(int k=0;k<5;k++){
				if(numero>0.5){
					//Genera un numero random de 0 a tipoCaracteristicas.length()-1
					numero2=Math.random()*fisicas[k].length;
					caracteristicas.add(fisicas[k][(int)numero2]);
				}else{
					//Genera un numero random de 0 a tipoCaracteristicas.length()-1
					numero2=Math.random()*sobresalientes[k].length;
					caracteristicas.add(sobresalientes[k][(int)numero2]);
				}
				
			}
			//System.out.println(Arrays.toString(caracteristicas.toArray()));
			this.personajes.put(Integer.toString(i),caracteristicas);
		}

	}

	public ArrayList<String> getGanador(){
		return personajes.get(Integer.toString(this.ganador));
	}
	/*public static void main(String args[]){
		Personajes lista=new Personajes();
		ArrayList<String> caracteristicasGanador=lista.getGanador();
		System.out.println("Lista de caracteristicas de los personajes: ");
		for (int i=0;i<10;i++){
			System.out.println("Personaje "+i+ ": "+Arrays.toString(lista.personajes.get(Integer.toString(i)).toArray()));
		}
		System.out.println("Lista de caracteristicas del Personaje ganador");
		System.out.println(Arrays.toString(caracteristicasGanador.toArray()));
	}*/
}

File: Problema1/Personajes.py
import random

class personajes:
    personajes={} #Diccionario de la forma Nombre Personaje: Características (Caracteristicas es un diccionario de la forma Tipo Caracteristica: Caracteristica)
    caracteristicasFisicas={}
    caracteristicasFisicas['pelo']=["Azul","Negro","Amarillo","Rojo","Cafe","Violeta","Gris","Verde","Plateado","Rosa"] #Color de pelo
    caracteristicasFisicas['complexion']=["Delgado","Flaco","Gordo","Morbido","Atletico","Robusto","Perfecto","Esqueletico","Normal","Amorfo"] #Complexión
    caracteristicasFisicas['estatura']=["Alto","Bajo","Enano","Gigante","Godzilla","Liliputiense","Microscopico","Planeta","Subatomico","Enorme"]#Estatura
    caracteristicasFisicas['piel']=["Blanco","Cafe","Negro","Rojo","Azul","Verde","Morado","Rosa","Amarillo","Carton"] #Color de piel
    caracteristicasFisicas['raza']=["Humano","Monstruo","Vampiro","Depredador","Robot","Reptiliano","Atlante","Demonio","Alien","Marciano"]#Diccionario constante con las caracteristicas fisicas posibles para los personajes (Tipo de Caracteristica: lista de caracteristicas)
    
    caracteristicasSobresalientes={}
    caracteristicasSobresalientes['arte']=["Arquitectura","Cine","Pintura","Escultura","Poesia","Musica","Danza"] #Artes
    caracteristicasSobresalientes['deporte']=["Futbol","Beisbol","Basquetbol","Automovilismo","Pesca","Arqueria"] #Deportes
    caracteristicasSobresalientes['ciencia']=["Quimica","Fisica","Biologia","Computacion","Economia","Epidemiologia","Farmacologia"] #/Ciencia
    caracteristicasSobresalientes['grado']=["Licenciado","Ingeniero","Maestro","Doctor","Empresario","Bachiller"] #Grado
    caracteristicasSobresalientes['pasatiempo']=["Rompecabezas","Acuarios","Terrarios","Legos","Caminata","Maquetas","Dormir"]#Diccionario constante con las caracteristicas sobresalientes posibles para los personajes (Tipo de Caracteristica: lista de caracteristicas)
	
    nombres=["Pedro", "Alex", "Ricardo", "Maria", "Sofia", "Karen", "Valentina", "Cecilia", "Diana", "Fidel"] #Lista constante con los nombres posibles para los personajes (10 nombres cualesquiera)
	
    correcto="" #Nombre del personaje a adivinar (se escoge al azar al terminar de llenar el diccionario de personajes)
    ganador=False
    ultimaJugada=[]
    turnos=[]
    def __init__(self):
        #Constructor de la clase personajes aquí se llenará el diccionario personajes al azar
        self.correcto=self.nombres[int(random.random()*len(self.nombres))]
        self.generaCaracteristicas()
        print(self.correcto)
	
    def generaCaracteristicas(self):
        numero=random.random()
        if numero<=0.7 and numero>=0.2:
            diccionario=self.caracteristicasFisicas
        else:
            diccionario=self.caracteristicasSobresalientes
        for i in range(0,10):
            caracteristicas={}
            for key in diccionario:
                numero2=int(random.random()*len(diccionario[key]))
                caracteristicas[key]=diccionario[key][numero2]
            numero2=int(random.random()*len(self.nombres))
            llave=self.nombres.pop(numero2)
            self.personajes[llave]=caracteristicas #aqui se añade el personaje con las carcateristicas generadas

    def getGanador(self):
        return (self.correcto,self.personajes[self.correcto])


